src/engine.ts

```diff
@@ -115,26 +115,46 @@ export class CSEngine extends Engine<never, CSTask> {
     }
   }
 
+  /**
+   * If we have reached an actual cost of 1 turn, no need to do more tasks
+   * @returns Next task or an early undefined if appropriate, stopping the task loop
+   */
+  getNextTask(): CSTask | undefined {
+    if (
+      this.csOptions.type === "SERVICE" &&
+      this.csOptions.test.prediction <= 1 &&
+      this.csOptions.test.actualCost() <= 1
+    ) {
+      return undefined;
+    }
+
+    return super.getNextTask();
+  }
+
   private get turns(): number {
     if (!this.turnsSpent) return 0;
     return undelay(this.turnsSpent);
   }
 
+  execute(task: CSTask): void {
+    withCheckpoint(() => super.execute(task));
```

Comment:
we should only wrap this in a checkpoint wrapper during an actual test, not during a non-test task (like leveling, where this will add a LOT of overhead for no tangible benefit)